client/main.go

```diff
@@ -465,7 +465,14 @@ func main() {
 					jitterHist.RecordValue(jitter)
 					globalJitterHist.RecordValue(jitter)
 				}
-				if *totalRequests > 0 && totalCount >= *totalRequests {
+				if *totalRequests > 0 && totalCount + 1 >= *totalRequests {
+					if resp.err != nil {
+						bad++
+						totalBad++
+					} else {
+						good++
+						totalGood++
+					}
```

Comment:
You make a good point. Apologies for not noticing that. If it's alright with you, I'll just close this pull request for now and reopen one if I find a solution to this concurrency issue. Thank you for the great feedback and have a great day!